ako-gateway-api/status/httproute_status.go

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2023-2024 VMware, Inc.
+ * All Rights Reserved.
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*   http://www.apache.org/licenses/LICENSE-2.0
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package status
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	akogatewayapilib "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/ako-gateway-api/lib"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/status"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pkg/utils"
+)
+
+type httproute struct{}
+
+func (o *httproute) Get(key string, name string, namespace string) *gatewayv1beta1.HTTPRoute {
+
+	obj, err := akogatewayapilib.AKOControlConfig().GatewayApiInformers().HTTPRouteInformer.Lister().HTTPRoutes(namespace).Get(name)
+	if err != nil {
+		utils.AviLog.Warnf("key: %s, msg: unable to get the HTTPRoute object. err: %s", key, err)
+		return nil
+	}
+	utils.AviLog.Debugf("key: %s, msg: Successfully retrieved the HTTPRoute object %s", key, name)
+	return obj.DeepCopy()
+}
+
+func (o *httproute) GetAll(key string) map[string]*gatewayv1beta1.HTTPRoute {
+
+	objs, err := akogatewayapilib.AKOControlConfig().GatewayApiInformers().HTTPRouteInformer.Lister().List(labels.Everything())
+	if err != nil {
+		utils.AviLog.Warnf("key: %s, msg: unable to get the HTTPRoute objects. err: %s", key, err)
+		return nil
+	}
+
+	httpRouteMap := make(map[string]*gatewayv1beta1.HTTPRoute)
+	for _, obj := range objs {
+		httpRouteMap[obj.Namespace+"/"+obj.Name] = obj.DeepCopy()
+	}
+
+	utils.AviLog.Debugf("key: %s, msg: Successfully retrieved the HTTPRoute objects", key)
+	return httpRouteMap
+}
+
+func (o *httproute) Delete(key string, option status.StatusOptions) {
+	// TODO: Add this code when we publish the status from the rest layer
+}
+
+func (o *httproute) Update(key string, option status.StatusOptions) {
+	// TODO: Add this code when we publish the status from the rest layer
```

Comment:
What scenarios will it cover? What are changes missing from rest layer?

Reply:
same as gatewayClass, status update from rest layer will go to gateways, these are placeholders for now.